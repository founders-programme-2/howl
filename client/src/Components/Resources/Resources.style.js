import styled from 'styled-components';

export const Sh2 = styled.h2`
  font-size: 1.3rem;
`;

export const Sp = styled.p`
  color: var(--main-font-color);
  margin: 0rem 2rem;
`;
