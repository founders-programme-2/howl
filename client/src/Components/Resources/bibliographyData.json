{
  "books": [
    {
      "title": "Here is a Title",
      "author": "Author's Name",
      "date": "Date of Publication",
      "description": "Some text describing the book"
    },
    {
      "title": "Here is a Title2",
      "author": "Author's Name2",
      "date": "Date of Publication2",
      "description": "Some text describing the book2"
    },
    {
      "title": "Sisterhood is Powerful",
      "author": "Robin Morgan, editor",
      "date": "1970-71",
      "description": "The first American feminist anthology"
    }
  ]
}
