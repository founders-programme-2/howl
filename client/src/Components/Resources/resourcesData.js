const resourcesData = [
  {
    id: 0,
    title: "British Library Timeline of the Women's Liberation Movement",
    link: 'https://www.bl.uk/sisterhood/timeline',
    info:
      'Listen to oral history interviews featuring activists of the Women’s Liberation Movement',
  },
  {
    id: 1,
    title: 'London School of Economics and Political Science',
    link:
      'http://www.lse.ac.uk/Library/Collections/Collection-highlights/The-Womens-Library',
    info: "Information on the Women's Library",
  },
  {
    id: 2,
    title: 'Feminist Library',
    link: 'http://feministlibrary.co.uk/',
    info:
      "The Feminist Library is a large collection of Women's Liberation Movement literature based in London.",
  },
  {
    id: 3,
    title: 'Bishopsgate Institute',
    link: 'https://www.bishopsgate.org.uk/Library',
    info:
      "Bishopsgate Institute's printed and archival collections cover a variety of subject areas exploring the experiences of everyday people, and including a special focus on feminist and women's history.",
  },
  {
    id: 4,
    title: 'Southall Black Sisters',
    link: 'https://southallblacksisters.org.uk/',
    info:
      'Southall Black Sisters, a not-for-profit, secular and inclusive organisation, was established in 1979 to meet the needs of Black (Asian and African-Caribbean) women. Their aims are to highlight and challenge all forms gender-related violence against women, empower them to gain more control over their lives; live without fear of violence and assert their human rights to justice, equality and freedom.',
  },
  {
    id: 5,
    title: 'Blog by Finn Mackay',
    link: 'https://finnmackay.wordpress.com/about/',
    info: 'Blog of feminist activist Finn Mackay',
  },
  {
    id: 6,
    title: "Glasgow Women's Library",
    link: 'https://womenslibrary.org.uk/',
    info:
      'Glasgow Women’s Library is the only Accredited Museum in the UK dedicated to women’s lives, histories and achievements, with a lending library, archive collections and innovative programmes of public events & learning opportunities.',
  },
  {
    id: 7,
    title: 'Feminist Archive North',
    link: 'https://feministarchivenorth.org.uk/',
    info:
      'The Women’s Liberation Movement was a major factor in social change, from the groups and demonstrations of the 1970s to the activist policy work of the 1980s and 1990s. The 21st century feminist revival is a strong and vibrant force today. In our collections you can find out where ideas began, and how feminist politics were put into practice. The Feminist Archive has the original sources.',
  },
  {
    id: 8,
    title: 'Feminist Archive South',
    link: 'http://feministarchivesouth.org.uk/about/',
    info:
      'THE FEMINIST ARCHIVE (SOUTH) IN BRISTOL, ENGLAND, HOUSES NATIONAL AND INTERNATIONAL MATERIAL RELATING TO THE HISTORY OF FEMINISM, C.1960-2000. IT IS the southern branch of the Feminist Archive.',
  },
];

export default resourcesData;
