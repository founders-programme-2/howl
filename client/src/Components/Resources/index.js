import React, { Fragment } from 'react';
import resourcesData from './resourcesData';
import bibliographyData from './bibliographyData';
import { Sh2, Sp } from './Resources.style';
import Footer from '../Common/Footer';

const Resources = () => {
  // maps over data and renders resources on page
  const renderResources = resourcesData.map(resource => {
    return (
      <article key={resource.id}>
        <a href={resource.link} target="_blank" rel="noopener noreferrer">
          <Sh2>{resource.title}</Sh2>
        </a>
        <Sp>{resource.info}</Sp>
      </article>
    );
  });

  // maps over books in data.json and renders bibliography
  const renderBibliography = bibliographyData.books.map(book => {
    return (
      <article key={book.title}>
        <Sh2>Title: {book.title}</Sh2>
        <Sp>Author: {book.author}</Sp>
        <Sp>Date: {book.date}</Sp>
        <Sp>Description: {book.description}</Sp>
      </article>
    );
  });
  return (
    <Fragment>
      <main>
        <header>
          <h1>Feminist Resources</h1>
        </header>
        <p>
          While HOWL exists to fill a gap in the archival and sharing of
          personal stories, many other thorough and valuable feminist resources
          exist. If you are looking to learn more, please have a look at the
          following:
        </p>
        {renderResources}
        <h2>Bibliography</h2>
        {renderBibliography}
      </main>
      <Footer />
    </Fragment>
  );
};

export default Resources;
