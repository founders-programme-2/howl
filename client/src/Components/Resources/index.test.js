import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Resources from './index';

test('Natural', () => {
  const resources = renderer.create(<BrowserRouter> <Resources /> </BrowserRouter>).toJSON();
  expect(resources).toMatchSnapshot();
});
