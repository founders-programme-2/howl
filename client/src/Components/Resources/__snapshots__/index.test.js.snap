// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Natural 1`] = `
Array [
  " ",
  <main>
    <header>
      <h1>
        Feminist Resources
      </h1>
    </header>
    <p>
      While HOWL exists to fill a gap in the archival and sharing of personal stories, many other thorough and valuable feminist resources exist. If you are looking to learn more, please have a look at the following:
    </p>
    <article>
      <a
        href="https://www.bl.uk/sisterhood/timeline"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          className="sc-bdVaJa gxzuMa"
        >
          British Library Timeline of the Women's Liberation Movement
        </h2>
      </a>
      <p
        className="sc-bwzfXH gYnxCV"
      >
        Listen to oral history interviews featuring activists of the Women’s Liberation Movement
      </p>
    </article>
    <article>
      <a
        href="http://www.lse.ac.uk/Library/Collections/Collection-highlights/The-Womens-Library"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          className="sc-bdVaJa gxzuMa"
        >
          London School of Economics and Political Science
        </h2>
      </a>
      <p
        className="sc-bwzfXH gYnxCV"
      >
        Information on the Women's Library
      </p>
    </article>
    <article>
      <a
        href="http://feministlibrary.co.uk/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          className="sc-bdVaJa gxzuMa"
        >
          Feminist Library
        </h2>
      </a>
      <p
        className="sc-bwzfXH gYnxCV"
      >
        The Feminist Library is a large collection of Women's Liberation Movement literature based in London.
      </p>
    </article>
    <article>
      <a
        href="https://www.bishopsgate.org.uk/Library"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          className="sc-bdVaJa gxzuMa"
        >
          Bishopsgate Institute
        </h2>
      </a>
      <p
        className="sc-bwzfXH gYnxCV"
      >
        Bishopsgate Institute's printed and archival collections cover a variety of subject areas exploring the experiences of everyday people, and including a special focus on feminist and women's history.
      </p>
    </article>
    <article>
      <a
        href="https://southallblacksisters.org.uk/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          className="sc-bdVaJa gxzuMa"
        >
          Southall Black Sisters
        </h2>
      </a>
      <p
        className="sc-bwzfXH gYnxCV"
      >
        Southall Black Sisters, a not-for-profit, secular and inclusive organisation, was established in 1979 to meet the needs of Black (Asian and African-Caribbean) women. Their aims are to highlight and challenge all forms gender-related violence against women, empower them to gain more control over their lives; live without fear of violence and assert their human rights to justice, equality and freedom.
      </p>
    </article>
    <article>
      <a
        href="https://finnmackay.wordpress.com/about/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          className="sc-bdVaJa gxzuMa"
        >
          Blog by Finn Mackay
        </h2>
      </a>
      <p
        className="sc-bwzfXH gYnxCV"
      >
        Blog of feminist activist Finn Mackay
      </p>
    </article>
    <article>
      <a
        href="https://womenslibrary.org.uk/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          className="sc-bdVaJa gxzuMa"
        >
          Glasgow Women's Library
        </h2>
      </a>
      <p
        className="sc-bwzfXH gYnxCV"
      >
        Glasgow Women’s Library is the only Accredited Museum in the UK dedicated to women’s lives, histories and achievements, with a lending library, archive collections and innovative programmes of public events & learning opportunities.
      </p>
    </article>
    <article>
      <a
        href="https://feministarchivenorth.org.uk/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          className="sc-bdVaJa gxzuMa"
        >
          Feminist Archive North
        </h2>
      </a>
      <p
        className="sc-bwzfXH gYnxCV"
      >
        The Women’s Liberation Movement was a major factor in social change, from the groups and demonstrations of the 1970s to the activist policy work of the 1980s and 1990s. The 21st century feminist revival is a strong and vibrant force today. In our collections you can find out where ideas began, and how feminist politics were put into practice. The Feminist Archive has the original sources.
      </p>
    </article>
  </main>,
  " ",
]
`;
